test/e2e/federatedcluster/clusterstatus.go

```diff
@@ -124,18 +125,10 @@ var _ = ginkgo.Describe("Cluster Status", federatedClusterTestLabels, func() {
 			}).WithTimeout(clusterStatusUpdateInterval*2).WithContext(ctx).Should(gomega.Succeed(), "Timed out waiting for status convergence")
 		}
 
-		makeClusterUnreachable := func(ctx context.Context) {
-			_, err := f.HostFedClient().
-				CoreV1alpha1().
-				FederatedClusters().
-				Patch(ctx, cluster.Name, types.JSONPatchType, invalidClusterEndpointPatchData, metav1.PatchOptions{})
-			gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
-		}
-
 		ginkgo.Context("Without service account", func() {
 			ginkgo.It("Should succeed", func(ctx ginkgo.SpecContext) {
 				ginkgo.By("Creating cluster")
-				cluster, _ = f.NewCluster(ctx, framework.WithTaints)
+				cluster, _, stopClusterFn = f.NewCluster(ctx, framework.WithTaints)
```

Comment:
Why do we return a `stopClusterFn` instead of letting the caller use `StopCluster` directly?

Reply:
I agree that `ListWatcher` is not the best name... But I feel `PaginatedReflector` is also not ideal, the name `Reflector` is based on the idea that a reflector "reflects" events into a store, which does happen here

Reply:
> I think both works, but the reason why I returned a stopClusterFn is because I was not sure how to extend the interface of Framework, specifically what type of argument should be used to identify the cluster to be stopped - name, uid, cluster object?
>
> Preface: StopCluster is currently only a method of the ClusterProvider interface, which is not directly exposed to users of the framework.

You replied this on the wrong thread... For `stopCluster`, I think it's reasonable to expose it on the `Framework`. I think taking the cluster name should be sufficient. WDYT?